src/Ng/Jobs/EndpointMonitoringJob.cs

```diff
@@ -71,10 +70,11 @@ protected override void Init(IDictionary<string, string> arguments, Cancellation
 
             var monitoringStorageFactory = CommandHelpers.CreateStorageFactory(arguments, verbose);
             var auditingStorageFactory = CommandHelpers.CreateSuffixedStorageFactory("Auditing", arguments, verbose);
-
-            var endpointNames = arguments.GetOrThrow<string>(Arguments.EndpointsToTest).Split(';');
-            var endpoints = endpointNames.Select(name => 
-                new EndpointFactory.Input(name, new Uri(arguments.GetOrThrow<string>(Arguments.EndpointCursorPrefix + name))));
+            
+            var endpoints = arguments.GetOrThrow<string>(Arguments.EndpointsToTest).Split(';').Select(e => {
+                var endpointParts = e.Split('|');
+                return new EndpointFactory.Input(endpointParts[0], new Uri(endpointParts[1]));
```

Comment:
Before it would throw if 'EndpointCursorPrefix + name' argument was missing, but now it would result in out of bounds exception. Maybe validate that split of '|' has the expected count and show usage if not.